manager-rs/dauth-service/src/rpc/clients/backup_network.rs

```diff
@@ -13,18 +27,98 @@ pub async fn enroll_backup_prepare(
     backup_network_id: &str,
     address: &str,
 ) -> Result<(), DauthError> {
-    todo!()
+    add_client(context.clone(), address).await?;
+
+    let sent_payload = enroll_backup_prepare_req::Payload {
+        home_network_id: context.local_context.id.clone(),
+        backup_network_id: backup_network_id.to_string(),
+        user_id_kind: UserIdKind::Supi as i32,
+        user_id: user_id.as_bytes().to_vec(),
+    };
+
+    let response = context
+        .rpc_context
+        .backup_clients
+        .lock()
+        .await
+        .get_mut(address)
+        .ok_or_else(|| DauthError::ClientError("Failed to get client".to_string()))?
+        .enroll_backup_prepare(EnrollBackupPrepareReq {
+            message: Some(signing::sign_message(
+                context.clone(),
+                SignPayloadType::EnrollBackupPrepareReq(sent_payload.clone()),
+            )),
+        })
+        .await?
+        .into_inner();
+
+    let message = response.message.ok_or(DauthError::ClientError(
+        "Missing signed message".to_string(),
+    ))?;
+
+    if let SignPayloadType::EnrollBackupPrepareReq(payload) =
+        signing::verify_message(context.clone(), &message).await?
+    {
+        if payload == sent_payload {
+            Ok(())
+        } else {
+            Err(DauthError::ClientError(format!(
+                "Failed to verify enroll backup contents: {:?}/s {:?}/r",
+                sent_payload, payload
+            )))
+        }
+    } else {
+        Err(DauthError::ClientError(format!(
+            "Incorrect message type received: {:?}",
+            message
+        )))
+    }
 }
 
 /// Send the set of initial vectors and key shares after
 /// a network has agreed to be a backup.
 pub async fn enroll_backup_commit(
     context: Arc<DauthContext>,
+    backup_network_id: &str,
+    user_id: &str,
     vectors: Vec<AuthVectorRes>,
     key_shares: Vec<(HresStar, Kseaf)>,
     address: &str,
 ) -> Result<(), DauthError> {
-    todo!()
+    let mut dvectors = Vec::new();
+    let mut dshares = Vec::new();
+
+    for vector in &vectors {
+        dvectors.push(build_delegated_vector(
+            context.clone(),
+            vector,
+            backup_network_id,
+        ))
+    }
+    for (xres_star_hash, confirmation_share) in &key_shares {
+        dshares.push(build_delegated_share(
```

Comment:
This part will be tricky, since the shares corresponding to this network's AUTHs should actually go to all the *other* backup networks not being enrolled, and this backup network will need to get key shares corresponding to the other backup networks. Synchronizing/updating the set of networks is hard, and part of why my mental model was that the entire set of backup networks should be known prior to enrolling any of them in the basic system. Updating the set requires generating new key shares and pushing the new key shares out to all networks.